src/main/java/org/klodnicki/controller/LendBookController.java

```diff
@@ -0,0 +1,68 @@
+import org.klodnicki.service.BookService;
+import org.klodnicki.service.LendBookService;
+
+public class LendBookController {
+
+    private final MenuController menuController;
+
+    private final LendBookService lendBookService = new LendBookService(new AccountService(), new BookService());
+
+    public LendBookController(MenuController menuController) {
+        this.menuController = menuController;
+    }
+
+    private static final String LEND_BOOK_PROCEDURE = "In order to lend a book please fill the following information. " +
+            "\nRemember that each reader can have maximum " + LendBookService.LENT_BOOK_LIMIT + " books borrowed at the same time.";
+    private static final String READER_INFORMATION = "Reader information:";
+    private static final String FIRST_NAME = "First name:";
+    private static final String LAST_NAME = "Last name:";
+    private static final String PESEL = "Pesel:";
+    private static final String BOOK_INFORMATION = "Book information:";
+    private static final String BOOK_TITLE = "Title of a book:";
+    private static final String AUTHOR = "Author:";
+    private static final String MORE_THAN_ONE_EDITION = "This book has more than one edition.";
+    private static final String EDITION_CHOICE = "Select edition of your choice:";
+    private static final String ABORT_OPERATION = "An operation has been canceled.";
+    private static final String LEND_BOOK_SUCCESS = "Success! The book has been successfully lent.";
+    private static final String NOT_FOUND = "No result found.";
+
+
+    public void lend() throws IllegalArgumentException, NoResultException {
+        menuController.displayOnMenu(LEND_BOOK_PROCEDURE);
+        menuController.displayOnMenu(READER_INFORMATION);
+        String firstName = menuController.displayOnMenuAndAskForInput(FIRST_NAME);
+        String lastName = menuController.displayOnMenuAndAskForInput(LAST_NAME);
+        String pesel = menuController.displayOnMenuAndAskForInput(PESEL);
+
+        menuController.displayOnMenu(BOOK_INFORMATION);
+        String title = menuController.displayOnMenuAndAskForInput(BOOK_TITLE);
+        String author = menuController.displayOnMenuAndAskForInput(AUTHOR);
+        String edition = null;
+
+        if (lendBookService.ifHasMoreThanOneEdition(title, author)) {
+            menuController.displayOnMenu(MORE_THAN_ONE_EDITION);
+            menuController.displayOnMenu(lendBookService.prepareListOfBooks(title, author));
+            edition = menuController.displayOnMenuAndAskForInput(EDITION_CHOICE);
+        }
+
+        try {
+            lendBookService.lend(firstName, lastName, pesel, title, author, edition);
+        } catch (NotEnoughBookCopiesException | MaximumBookBorrowedLimitException | NotFoundInDatabaseException e) {
+            menuController.displayOnMenu(e.getMessage());
+            menuController.displayOnMenu(ABORT_OPERATION);
+            return;
+        } catch (NoResultException e) {
```

Comment:
You shouldn't be catching this exception here! I believe you already handling this in the repository method where it actually occurs due to the `getSingleResult` method. Please, clear that one and I believe we could proceed.